# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/ahmedshereef1/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

Lane: Refresh / Content Opportunity Scoring

I chose the Refresh / Content Opportunity Scoring lane because it focuses on helping content teams prioritize which pages should be reviewed first. Instead of reviewing every page manually, the project will rank pages based on observable search and engagement signals. This makes the review process more efficient and helps teams spend their time on the pages with the highest potential impact.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

Research Question

Which content pages should be reviewed first for content refresh based on their search performance and engagement signals?

Decision

Determine which pages deserve immediate review.

Action

A content manager reviews the highest-ranked pages and decides whether to:

Update the content
Improve SEO
Expand the article
Monitor the page
Leave it unchanged
Unit of Analysis

One content page.

Output

A ranked list of pages with:

Refresh score
Confidence score
Reason codes explaining the recommendation
Cost of a Wrong Recommendation

If a page is incorrectly recommended for review, time and resources are wasted.

If an important page is missed, the business may lose valuable search traffic and potential customers.

Why Machine Learning?

Simple rules cannot capture the complex relationships between search visibility, user engagement, freshness, and content quality. Machine learning can combine multiple signals to produce better rankings than fixed rule-based systems while still providing explainable recommendations.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [1]:
import pandas as pd

df = pd.read_csv("../../data/raw/content_refresh_anonymized.csv")

print("Rows:", len(df))
print("Columns:", len(df.columns))
print("Unique content pages:", df["content_id"].nunique())

print(df["trend_direction"].value_counts())

Rows: 30000
Columns: 44
Unique content pages: 30000
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64


The starter dataset contains 30,000 rows and 44 features, representing 30,000 unique content pages. The trend_direction distribution shows 16,262 declining pages, 5,962 stable pages, 4,388 improving pages, 2,236 new pages, and 1,152 flat pages. These statistics indicate that the dataset contains a diverse range of content performance patterns, making it suitable for investigating which pages should be prioritized for review in a content refresh workflow.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

This project provides decision support, not automatic decision making.

The model does not prove that refreshing a page will increase traffic. It only identifies pages that appear to be good candidates for review based on historical search and engagement signals.

The recommendations should always be reviewed by a human before any action is taken.

## Self-check

Before you submit, confirm each line honestly:

- [x] Every section above is filled — markdown thinking AND the code that backs it
- [x] The notebook runs top to bottom with no errors (Runtime → Run all)
- [x] No client names, URLs, or private queries anywhere
- [x] My claims use careful words: observed, measured, directional, decision-support
- [x] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.